go2ts.go

```diff
@@ -75,19 +81,45 @@ func (g *Go2TS) Render(w io.Writer) error {
 	if err != nil {
 		return err
 	}
-	for _, st := range g.structs {
-		if err := st.render(w); err != nil {
+	for _, intf := range g.interfaces {
+		if err := intf.render(w); err != nil {
+			return err
+		}
+	}
+
+	for _, typ := range g.types {
+		if _, err := fmt.Fprintln(w, typ.String()); err != nil {
 			return err
 		}
 	}
+
 	return nil
 }
 
-// populateTypeDetails fills out the 'partialType' for the given 'typ'.
-//
-// It will recursively add subTypes to the partialType until the type is
-// completely described.
-func (g *Go2TS) tSTypeFromStructFieldType(reflectType reflect.Type) *tsType {
+// The list if primitive Kinds that we support converting to TypeScript.
+var primitive = map[reflect.Kind]bool{
+	reflect.Bool:    true,
+	reflect.Int:     true,
+	reflect.Int8:    true,
+	reflect.Int16:   true,
+	reflect.Int32:   true,
+	reflect.Int64:   true,
+	reflect.Uint:    true,
+	reflect.Uint8:   true,
+	reflect.Uint16:  true,
+	reflect.Uint32:  true,
+	reflect.Uint64:  true,
+	reflect.Uintptr: true,
+	reflect.Float32: true,
+	reflect.Float64: true,
+	reflect.String:  true,
+}
+
+func isPrimitive(kind reflect.Kind) bool {
+	return primitive[kind]
+}
+
+func (g *Go2TS) tsTypeFromReflectType(reflectType reflect.Type, calledFromAddType bool) *tsType {
```

Comment:
LOL, I changed this at the last minute from isRecursive to calledFromAddType because I thought isRecusive was too oblique.

Changed back to isRecusive.